src/components/Caption/CaptionElements.jsx

```diff
@@ -0,0 +1,64 @@
+import "./CaptionElements.css";
+import React, { useState } from "react";
+import { FaPen } from "react-icons/fa";
+
+const StyleChanger = (props) => {
+  const [cbState, setCbState] = useState(false);
+  const [penClicked, setPenClicked] = useState(false);
+
+  const changeHandler = (event) => {
+    setCbState(event.target.checked);
+    props.onCbValue(!cbState);
+  };
+
+  const penClickHandler = () => {
+    setCbState(false);
+    props.onCbValue(false);
+
+    setPenClicked(true);
+    props.onPenClicked(true);
+  };
+
+  const saveClickHandler = () => {
+    setPenClicked(false);
+    props.onPenClicked(false);
+
+    props.onSaveClick();
+  };
+
+  const cancelClickHandler = () => {
+    setPenClicked(false);
+    props.onPenClicked(false);
+
+    props.onCancelClick();
+  };
+
+  return (
+    <div>
+      <FaPen
+        onClick={penClickHandler}
+        className={penClicked ? "no-display" : ""}
+      />
+      <input
+        type="checkbox"
+        onChange={changeHandler}
+        checked={cbState}
+        className={penClicked ? "no-display" : ""}
+      ></input>
+      <button
+        onClick={saveClickHandler}
+        className={penClicked ? "" : "no-display"}
+      >
+        Save
+      </button>
+      <button
+        onClick={cancelClickHandler}
+        className={penClicked ? "" : "no-display"}
+      >
+        Cancel
+      </button>
+    </div>
+  );
```

Comment:
1. Better use icons for save/cancel actions rather than buttons - more user-friendly.
2. One of the main principles of React is not to render anything should not be available. So better render required elements on condition rather then hide it by setting `display` attribute.